tests/components/unifi/test_sensor.py

```diff
@@ -719,31 +740,79 @@ async def test_wlan_client_sensors(
+            "201.683",
+        ),
+    ],
+)
 async def test_outlet_power_readings(
-    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker, mock_unifi_websocket
+    hass: HomeAssistant,
+    aioclient_mock: AiohttpClientMocker,
+    sensor: str,
+    expected_unique_id: str,
+    expected_value: any,
 ) -> None:
     """Test the outlet power reporting on PDU devices."""
     await setup_unifi_integration(hass, aioclient_mock, devices_response=[PDU_DEVICE_1])
 
-    assert len(hass.states.async_all()) == 5
-    assert len(hass.states.async_entity_ids(SENSOR_DOMAIN)) == 1
+    assert len(hass.states.async_all()) == 7
+    assert len(hass.states.async_entity_ids(SENSOR_DOMAIN)) == 3
 
     ent_reg = er.async_get(hass)
-    ent_reg_entry = ent_reg.async_get("sensor.dummy_usp_pdu_pro_outlet_2_outlet_power")
-    assert ent_reg_entry.unique_id == "outlet_power-01:02:03:04:05:ff_2"
+    ent_reg_entry = ent_reg.async_get(f"sensor.{sensor}")
+    assert ent_reg_entry.unique_id == expected_unique_id
     assert ent_reg_entry.entity_category is EntityCategory.DIAGNOSTIC
 
-    outlet_2 = hass.states.get("sensor.dummy_usp_pdu_pro_outlet_2_outlet_power")
-    assert outlet_2.attributes.get(ATTR_DEVICE_CLASS) == SensorDeviceClass.POWER
-    assert outlet_2.state == "73.827"
+    sensor_data = hass.states.get(f"sensor.{sensor}")
+    assert sensor_data.attributes.get(ATTR_DEVICE_CLASS) == SensorDeviceClass.POWER
+    assert sensor_data.state == expected_value
 
-    # Verify state update
-    pdu_device_state_update = deepcopy(PDU_DEVICE_1)
 
-    pdu_device_state_update["outlet_table"][1]["outlet_power"] = "123.45"
+@pytest.mark.parametrize(
+    ("sensor", "changed_data", "expected_value"),
+    [
+        (
+            "dummy_usp_pdu_pro_outlet_2_outlet_power",
+            {"outlet_table": PDU_OUTLETS_UPDATE_DATA},
+            "123.45",
+        ),
+        (
+            "dummy_usp_pdu_pro_ac_power_consumption",
+            {"outlet_ac_power_consumption": "456.78"},
+            "456.78",
+        ),
+    ],
+)
+async def test_outlet_power_readings_update(
+    hass: HomeAssistant,
+    aioclient_mock: AiohttpClientMocker,
+    mock_unifi_websocket,
+    sensor: str,
+    changed_data: dict,
+    expected_value: any,
+) -> None:
+    """Tests that sensor updates are reflected in the sensor data."""
+    await setup_unifi_integration(hass, aioclient_mock, devices_response=[PDU_DEVICE_1])
+    updated_device_data = deepcopy(PDU_DEVICE_1)
+    updated_device_data.update(changed_data)
 
-    mock_unifi_websocket(message=MessageKey.DEVICE, data=pdu_device_state_update)
+    mock_unifi_websocket(message=MessageKey.DEVICE, data=updated_device_data)
     await hass.async_block_till_done()
 
-    outlet_2 = hass.states.get("sensor.dummy_usp_pdu_pro_outlet_2_outlet_power")
-    assert outlet_2.state == "123.45"
+    sensor_data = hass.states.get(f"sensor.{sensor}")
+    assert sensor_data.state == expected_value
```

Comment:
Combine the tests into one, no need to test updating separately. This should be the "final" comment :)